proxy/proxy.go

```diff
@@ -1175,6 +1175,8 @@ func (p *Proxy) serveResponse(ctx *context) {
 		p.tracing.setTag(ctx.proxySpan, ClientRequestStateTag, ClientRequestCanceled)
 	}
 
+	p.tracing.setTag(ctx.initialSpan, HTTPStatusCodeTag, uint16(ctx.response.StatusCode))
```

Comment:
This is a successful response branch of
https://github.com/zalando/skipper/blob/ff61a38328ee6160111dee40f935d0bd239b297c/proxy/proxy.go#L1426-L1432
but there is also a error branch (e.g. skipper failed to connect to backend). Should the  error branch also be covered?

Reply:
@AlexanderYastrebov error should be already covered by this https://github.com/zalando/skipper/blob/master/proxy/proxy.go#L1225, inside `errorResponse`

Reply:
Could you rewrite it then to explicit `initialSpan` since you are introducing it?

Reply:
@AlexanderYastrebov I am not 100% sure that they are interchangeable? Are they? `spanFromContext` is getting the currently "active span", it might be some other span eventually?

Reply:
From
https://github.com/zalando/skipper/blob/ff61a38328ee6160111dee40f935d0bd239b297c/proxy/proxy.go#L1411-L1413
https://github.com/zalando/skipper/blob/ff61a38328ee6160111dee40f935d0bd239b297c/proxy/context.go#L131-L151
https://github.com/opentracing/opentracing-go/blob/3088eee7e4d26010ac9a301e977fd2721dbebcbe/gocontext.go#L26-L32
I think 
https://github.com/zalando/skipper/blob/76fb0a7705c6238d6be1c3e441521bdb44752006/proxy/proxy.go#L1225-L1232
will get initial `span` and I think it is done like that because it was not passed around in the `ctx`.

Also from the use of initial span it looks like it is assumed to be non-nil so the nil check seems to be redundant.

Reply:
@AlexanderYastrebov applied the requested change